src/mark.js

```diff
@@ -157,23 +158,30 @@ class Mark { // eslint-disable-line no-unused-vars
      */
     createSynonymsRegExp(str) {
         const syn = this.opt.synonyms,
-            sens = this.opt.caseSensitive ? "" : "i";
+            sens = this.opt.caseSensitive ? "" : "i",
+            // add replacement character placeholder before and after the
+            // synonym group
+            joinerPlaceholder = this.opt.ignoreJoiners ||
+                this.opt.ignorePunctuation.length ? "\u0000" : "";
```

Comment:
@Mottie As this check wasn't there previously for the ignoreJoiners option, does this mean that the option was working incorrectly when using synonyms?